src/annotator/config/config.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+var settings = require('./settings');
+var sharedSettings = require('../../shared/settings');
+
+/**
+ * Reads the Hypothesis configuration from the environment.
+ *
+ * @param {Window} window_ - The Window object to read config from.
+ */
+function configFrom(window_) {
+  var config = {};
+
+  config.app = settings.iFrameSrc(window_.document);
+
+  Object.assign(config, sharedSettings.jsonConfigsFrom(window_.document));
+  Object.assign(config, settings.configFuncSettingsFrom(window_));
+
+  // Convert legacy keys/values in config to corresponding current
+  // configuration.
+  if (typeof config.showHighlights === 'boolean') {
+    config.showHighlights = config.showHighlights ? 'always' : 'never';
+  }
+
+  // Extract the default annotation ID or query from the URL.
+  //
+  // The Chrome extension or proxy may already have provided this config
+  // via a tag injected into the DOM, which avoids the problem where the page's
+  // JS rewrites the URL before Hypothesis loads.
```

Comment:
This is another thing that I can tidy up once the `Object.assign`s above are gone.

At this line in the code the `config.query` and `config.annotations` settings _may_ have already been set by the `Object.assign`s above. The intention is that if they have _not_ been set then these two lines of code below set them from the URL.

So the logic for how we read the `config.query` setting is spread throughout `jsonConfigsFrom()` (in which `query` is never mentioned by name), `configFuncSettingsFrom()` (`query` is never mentioned by name there either), and right here.

What I want to get to is the `settings.query()` function will encapsulate all of that logic in one place - it will look for `query` in js-hypothesis-configs and if it's not there will look for it in the URL, and return it.